bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/renault/api/Car.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.renault.internal.renault.api;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ * MyRenault registered car for parsing HTTP responses and collecting data and
+ * information.
+ * 
+ * @author Doug Culnane - Initial contribution
+ */
+@NonNullByDefault
+public class Car {
+
+    private final Logger logger = LoggerFactory.getLogger(Car.class);
+
+    public Double batteryLevel = Double.valueOf(-1);
+    public Boolean hvacstatus = false;
+    public Double odometer = Double.valueOf(0);
+    public String imageURL = "";
+    public Double gpsLatitude = Double.valueOf(0);
+    public Double gpsLongitude = Double.valueOf(0);
```

Comment:
I do not understand why you are now setting a default value, non null. Because in your thing handler (method updateState), the null value is your trigger to update the channel.
This look inconsistent to me.
You should probably add `@Nullable`

Reply:
I did this because I do not really know how to use the @NonNullByDefault properly.  Setting @Nullable caused a compile error for this code...

`        if (car.odometer != null) {
            updateState(CHANNEL_ODOMETER, new QuantityType<Length>(car.odometer, KILO(METRE)));
        }
`

required 'java.lang.@NonNull Number' but this expression has type 'java.lang.@Nullable Double'


I would like to use plain old Java and have the car values starting as null and then set properly.  I do not want defaults and I only want to update status if the value is set properly (not NULL).  However the SAT report say use the @NonNullByDefault on the car.  I would rather not or I need to know the proper trick here?

Reply:
Got a fix for this.  I will push @Nullable code. I am starting to understand better now.